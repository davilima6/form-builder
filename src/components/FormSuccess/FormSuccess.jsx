// @flow
import * as React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

import { useScrollToTop } from '../../utils';

const StyledFormSucess = styled.form`
  background-color: var(--lite-blue);
  height: 100vh;
  padding: 1em;
  text-align: center;
`;

const StyledButton = styled.button`
  border: 2px solid #000;
  border-radius: 3px;
  color: #000;
  cursor: pointer;
  display: inline-block;
  font-size: 1em;
  font-weight: bold;
  margin-top: 2em;
  padding: 0.5em 1em;
  text-decoration: none;

  :hover {
    background-color: var(--dark-green);
    color: #fff;
  }
`;

const FormSuccess = () => {
  useScrollToTop();

  return (
    <StyledFormSucess>
      <h2>Success!</h2>
      <p>Your data has been validated and sent to our team.</p>
      <p>Thank you for your submission.</p>
      <StyledButton as={Link} to="/">
        Try it again!
      </StyledButton>
    </StyledFormSucess>
  );
};

export default FormSuccess;
