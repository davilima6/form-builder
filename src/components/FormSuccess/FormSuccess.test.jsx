import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { cleanup, render } from 'react-testing-library';
import 'jest-dom/extend-expect';
// import 'jest-styled-components';
import FormSuccess from './FormSuccess';

describe('Form', () => {
  let component;

  beforeEach(() => {
    cleanup();
    window.scrollTo = () => {};
    component = render(
      <BrowserRouter>
        <FormSuccess />
      </BrowserRouter>,
    );
  });

  test('renders without crashing', () => expect(component).toBeDefined());

  test.skip('matches snapshot', () => {
    const { asFragment } = component;

    expect(asFragment()).toMatchSnapshot();
  });
});
