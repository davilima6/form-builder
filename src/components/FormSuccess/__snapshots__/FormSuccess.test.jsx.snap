// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="sc-bdVaJa toVoZ"
  >
    <h2>
      Success!
    </h2>
    <p>
      Your data has been validated and sent to our team.
    </p>
    <p>
      Thank you for your submission.
    </p>
    <a
      class="sc-bwzfXH oGNiN"
      href="/"
    >
      Try it again!
    </a>
  </form>
</DocumentFragment>
`;
